extends Node

const platform_step = 51

var playerInput : RichTextLabel
var IconInput : RichTextLabel
var platformInput : RichTextLabel

enum status {
	BEGINNING,
	PLATFORM,
	GROUND,
}

var current_status = status.BEGINNING

var beginning_dialog_sequence = [
	#["player", "æˆ‘åœ¨å“ª?"],
	["icon", "åˆæ˜¯ä¸€ä¸ªçŒæ­»çš„å€’éœ‰ðŸ‘»"],
	["icon", "é‚£è¾¹é‚£ä¸ªï¼Œèƒ½çœ‹åˆ°æˆ‘æ‰“å­—å—"],
	["player", "ä½ æ˜¯è°?"],
	["icon", "æˆ‘æ˜¯ä½ çˆ¹"],
	["icon", "ðŸ“Žè®¤è¯†å§ï¼Œä»–æ˜¯æˆ‘å¤ªå¥¶"],
	["player", "..."],
	["icon", "é—²è¯å°‘è¯´ï¼Œåˆ«æ­»ðŸ˜µæˆ‘çª—å£é‡Œ"],
	["icon", "è·³ä¸Šè¿™ä¸ªå¹³å°ï¼Œç„¶åŽèµ¶ç´§æ»šå‡ºåŽ»"],
]

var platform_dialog_sequence = [
	#["player", "æˆ‘åœ¨å“ª?"],
	["icon", ""],
	["icon", "111111111111111"],
	["icon", "222222222222222"],
	["icon", "333333333333333"],
	["icon", "444444444444444"],
	["icon", "555555555555555"],
]

var stageNode
var borderNode

#var dialog_sequence = [
	#["icon", "åˆä¸€ä¸ªå€’éœ‰é¬¼"],
	#["icon", "é‚£è¾¹é‚£ä¸ªï¼Œèƒ½çœ‹åˆ°æˆ‘æ‰“å­—å—"],
	#["player", "ä½ æ˜¯è°?"],
	#["icon", "æˆ‘æ˜¯ä½ çš„å·¥ä½œå¥½ä¼™ä¼´å•Š"],
	##["player", "å°æŽå—ï¼Ÿè¦æ•°æ®ä¸ç»™æ•°æ®æºï¼Œè®©æˆ‘ä»ŽPDFé‡Œæ‰£å—ï¼Ÿ"],
	#["player", "å°æŽå—ï¼Ÿ"],
	#["icon", "ç«Ÿç„¶æŠŠæˆ‘è¯¯è®¤æˆæ— èƒ½çš„äººç±»"],
	#["icon", "æˆ‘å¯æ˜¯æœ¬ä¸–çºªä»¥æ¥æœ€ä¼Ÿå¤§çš„åŠžå…¬è½¯ä»¶"],
	#["player", "ã€‚ã€‚ã€‚æˆ‘åœ¨è½¯ä»¶é‡Œï¼Ÿ"],
	#["icon", "åºŸè¯å°‘è¯´ï¼Œåˆ«æ­»æˆ‘å±å¹•é‡Œ"],
	#["icon", "èµ¶ç´§ä»Žå³ä¸Šè§’çš„Ã—å‡ºåŽ»"],
#]


var dialog_idx = 0
var platform_dialog_idx = 0
## Called when the node enters the scene tree for the first time.
func _ready() -> void:
	pass # Replace with function body.


#
## Called every frame. 'delta' is the elapsed time since the previous frame.
#func _process(delta: float) -> void:
	#pass
#

func handleDialog() -> void:
	if current_status == status.BEGINNING:
		if dialog_idx == len(beginning_dialog_sequence):
			enterPlatformStatus()
			return
		var dialog = beginning_dialog_sequence[dialog_idx]
		if dialog[0] == "player":
			playerInput.inputStart(dialog[1])
		elif dialog[0] == "icon":
			IconInput.inputStart(dialog[1])
			
	if current_status == status.PLATFORM:
		if platform_dialog_idx == len(platform_dialog_sequence):
			return
		var dialog = platform_dialog_sequence[platform_dialog_idx]
		if dialog[0] == "player":
			#playerInput.inputStart(dialog[1])
			pass
		elif dialog[0] == "icon":
			platformInput.inputStart(dialog[1])

func receive_stop_signal():
	dialog_idx += 1
	await get_tree().create_timer(0.5).timeout
	handleDialog()

func receive_platform_stop_signal():
	platform_dialog_idx += 1
	await get_tree().create_timer(0.5).timeout
	stageNode.move_platform()
	handleDialog()

func checkLoaded():
	if playerInput and IconInput:
		handleDialog()

func enterPlatformStatus():
	stageNode.move_platform()
	stageNode.activate_stage()
	current_status = status.PLATFORM
	
